hledger-core.el

```diff
@@ -55,9 +55,9 @@
   "Regular expression for matching a starting entry with some description.")
 
 (defvar hledger-account-regex
-  (concat "\\(\\([Rr]evenues?\\|[aA]ssets?\\|[lL]iabilit\\(?:ies\\|y\\)\\|[Dd]ebts?"
+  (concat "\\(\\(?:[Rr]evenues?\\|[aA]ssets?\\|[lL]iabilit\\(?:ies\\|y\\)\\|[Dd]ebts?"
 	  "\\|[Ee]quity\\|[Ee]xpenses?\\|[iI]ncome\\|[Zz]adjustments?\\)" ;; terms all accounts must start with
-	  "\\(:[A-Za-z0-9\-]+\\( [A-Za-z0-9\-]+\\)*\\)*\\)"             ;; allow multi-word account names
+	  "\\(?::\\([^[:space:];\n]+\\(?: [^[:space:];=\n]+\\)*\\)\\)*\\)"             ;; allow multi-word account names
```

Comment:
Just to make sure I understand this correctly: 

If there are two spaces or more between the account name and the amount in an entry, we will still correctly separate the two and font-lock will highlight them correctly. Right?

Reply:
Yes, that’s correct. `expenses:foo $42` is parsed as a single account named `expenses:foo $42`, whereas `expenses:foo  $42` (two spaces or more) is parsed as an account `expenses:foo` and amount `$42`.